test/functional/mempool_spend_coinbase.py

```diff
@@ -13,44 +13,49 @@
 """
 
 from test_framework.test_framework import BitcoinTestFramework
-from test_framework.blocktools import create_raw_transaction
 from test_framework.util import assert_equal, assert_raises_rpc_error
+from test_framework.wallet import MiniWallet
 
 
 class MempoolSpendCoinbaseTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
+        self.setup_clean_chain = True
 
     def run_test(self):
+        wallet = MiniWallet(self.nodes[0])
+
+        wallet.generate(200)
         chain_height = self.nodes[0].getblockcount()
         assert_equal(chain_height, 200)
-        node0_address = self.nodes[0].getnewaddress()
 
         # Coinbase at height chain_height-100+1 ok in mempool, should
         # get mined. Coinbase at height chain_height-100+2 is
-        # is too immature to spend.
+        # too immature to spend.
         b = [self.nodes[0].getblockhash(n) for n in range(101, 103)]
         coinbase_txids = [self.nodes[0].getblock(h)['tx'][0] for h in b]
-        spends_raw = [create_raw_transaction(self.nodes[0], txid, node0_address, amount=49.99) for txid in coinbase_txids]
+        utxo_101 = next(filter(lambda utxo: coinbase_txids[0] == utxo['txid'], wallet._utxos))
+        utxo_102 = next(filter(lambda utxo: coinbase_txids[1] == utxo['txid'], wallet._utxos))
```

Comment:
_utxos is a private member, so you can't read it directly. It needs to be accessed via a getter method that also marks the utxo as spent (pops it)

Reply:
Good point. What do you think of this approach? https://github.com/bitcoin/bitcoin/pull/20385/commits/650ba88013ff0fb8cf08b00d5e7c674cf5b16a22